File: test_data/code/java/1640-check-array-formation-through-concatenation.java
class Solution {
    public boolean canFormArray(int[] arr, int[][] pieces) {
        for (int[] piece : pieces) {
            boolean formed = true;
            int i = 0;
            while (i < arr.length) {
                if (formed) {
                    if (i + piece.length > arr.length) {
                        return false;
                    }
                    boolean match = true;
                    for (int j = 0; j < piece.length; j++) {
                        if (arr[i + j] != piece[j]) {
                            match = false;
                            break;
                        }
                    }
                    if (!match) {
                        return false;
                    }
                }
                i += piece.length;
            }
        }
        return true;
    }
}

File: test_data/code/python/1640-check-array-formation-through-concatenation.py
from typing import List

class Solution:
    def canFormArray(self, arr: List[int], pieces: List[List[int]]) -> bool:
        piece_index = 0
        for i in range(len(arr)):
            for piece in pieces:
                if arr[i:i + len(piece)] == piece:
                    piece_index += len(piece)
                    i += len(piece) - 1
                    break
            else:
                return False
        return True